coderd/prebuilds/state_test.go

```diff
@@ -0,0 +1,544 @@
+package prebuilds_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/quartz"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/prebuilds"
+)
+
+type options struct {
+	templateID        uuid.UUID
+	templateVersionID uuid.UUID
+	presetID          uuid.UUID
+	presetName        string
+	prebuildID        uuid.UUID
+	workspaceName     string
+}
+
+// templateID is common across all option sets.
+var templateID = uuid.New()
+
+const (
+	backoffInterval = time.Second * 5
+
+	optionSet0 = iota
+	optionSet1
+	optionSet2
+)
+
+var opts = map[uint]options{
+	optionSet0: {
+		templateID:        templateID,
+		templateVersionID: uuid.New(),
+		presetID:          uuid.New(),
+		presetName:        "my-preset",
+		prebuildID:        uuid.New(),
+		workspaceName:     "prebuilds0",
+	},
+	optionSet1: {
+		templateID:        templateID,
+		templateVersionID: uuid.New(),
+		presetID:          uuid.New(),
+		presetName:        "my-preset",
+		prebuildID:        uuid.New(),
+		workspaceName:     "prebuilds1",
+	},
+	optionSet2: {
+		templateID:        templateID,
+		templateVersionID: uuid.New(),
+		presetID:          uuid.New(),
+		presetName:        "my-preset",
+		prebuildID:        uuid.New(),
+		workspaceName:     "prebuilds2",
+	},
+}
+
+// A new template version with a preset without prebuilds configured should result in no prebuilds being created.
+func TestNoPrebuilds(t *testing.T) {
+	t.Parallel()
+	current := opts[optionSet0]
+	clock := quartz.NewMock(t)
+
+	presets := []database.GetTemplatePresetsWithPrebuildsRow{
+		preset(true, 0, current),
+	}
+
+	state := prebuilds.NewReconciliationState(presets, nil, nil, nil)
+	ps, err := state.FilterByPreset(current.presetID)
+	require.NoError(t, err)
+
+	actions, err := ps.CalculateActions(clock, backoffInterval)
+	require.NoError(t, err)
+
+	validateActions(t, prebuilds.ReconciliationActions{ /*all zero values*/ }, *actions)
```

Comment:
Do we need `validateActions` function? Can we just use smth like: `require.Equal()`?

Reply:
Moved